src/core/settingmanager.cpp

```diff
@@ -143,6 +157,74 @@ void SettingManager::doNotTrack()
                                         m_doNotTrackConfItem->value(false));
 }
 
+void SettingManager::handleObserve(const QString &message, const QVariant &data)
+{
+    const QVariantMap dataMap = data.toMap();
+    if (message == QLatin1String("embed:search")) {
+        QString msg = dataMap.value("msg").toString();
+        if (msg == QLatin1String("init")) {
+            const StringMap configs(OpenSearchConfigs::getAvailableOpenSearchConfigs());
+            const QStringList configuredEngines = configs.keys();
+            QStringList registeredSearches(dataMap.value(QLatin1String("engines")).toStringList());
+            QString defaultSearchEngine = dataMap.value(QLatin1String("defaultEngine")).toString();
+
+            m_searchEnginesInitialized = !registeredSearches.isEmpty();
+
+            // Upon first start, engine doesn't know about the search engines.
+            // Engine load requests are send within the for loop below.
+            if (!m_searchEnginesInitialized) {
+                m_addedSearchEngines = new QStringList(configuredEngines);
```

Comment:
Could there be a memory leak if search engine initialization fails while handling ```search-engine-added``` message ?  ```m_addedSearchEngines``` is deleted only if search engine initialization is succesful.